src/files.js

```diff
@@ -160,6 +182,52 @@ module.exports = (common) => {
           })
         })
 
+        it('add a nested dir as array with progress', (done) => {
+          // Needs https://github.com/ipfs/js-ipfs-api/issues/339 to be fixed
+          // for js-ipfs-api + go-ipfs
+          if (!isNode) { return done() }
+
+          const content = (name) => ({
+            path: `test-folder/${name}`,
+            content: directoryContent[name]
+          })
+
+          const emptyDir = (name) => ({
+            path: `test-folder/${name}`
+          })
+
+          const expectedRootMultihash = 'QmVvjDy7yF7hdnqE8Hrf4MHo5ABDtb5AbX6hWbD3Y42bXP'
+
+          const dirs = [
+            content('pp.txt'),
+            content('holmes.txt'),
+            content('jungle.txt'),
+            content('alice.txt'),
+            emptyDir('empty-folder'),
+            content('files/hello.txt'),
+            content('files/ipfs.txt'),
+            emptyDir('files/empty')
+          ]
+
+          let progCount = 0
+          let progress = 0
+          const handler = (p) => {
+            progCount += 1
+            progress = p
+          }
+
+          ipfs.files.add(dirs, {progress: handler}, (err, res) => {
+            expect(err).to.not.exist()
+            const root = res[res.length - 1]
+
+            expect(root.path).to.equal('test-folder')
+            expect(root.hash).to.equal(expectedRootMultihash)
+            expect(progCount).to.equal(8)
+            expect(progress).to.equal(5)
```

Comment:
This has to be wrong. How can a nested dir full of files have a `progress` of 5 bytes only?

Reply:
That is indeed wrong, a fix is coming.